### PR TITLE
[Integration Tests] Implement coverage for API/RPC

### PR DESCRIPTION
**Pull request by thefazzer**
_Added tests for voids (where API/RPC call controllers but these calls have no coverage)._
- StopStaking_Returns_Ok
- StartStaking_OnProofOfWorkNetwork_Returns_MethodNotallowed
- MiningController_Calls_GetAccount_Returns_Reference_To_Wallet_Account
- MiningController_Calls_GetAccount_Error_No_Account_Found
- MiningController_Calls_GetAccount_Error_No_Wallet_Found_

Preferred: @fassadlr @bokobza

### CODE REVIEW

**Comment by fassadlr** [comment]
@bokobza maybe you should take this one?

**Comment by bokobza** [comment]
yep, on it like a car bonnet.

File: src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/StakingControllerTest.cs [hunk 1]
```diff
@@ -210,5 +213,54 @@ public void StartStaking_InvalidTimeSyncState_ReturnsBadRequest()
...
+                Assert.IsType<OkResult>(startStakingResult);
+
+                IActionResult stopStakingResult = this.controller.StopStaking();
+
+                Assert.IsType<OkResult>(stopStakingResult);
+            }
+        }
+
+        [Fact]
+        public void StartStaking_OnProofOfWorkNetwork_Returns_MethodNotallowed()
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/StakingControllerTest.cs]:
```suggestion
        public void StartStaking_OnProofOfWorkNetwork_Returns_MethodNotAllowed()
```

File: src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs [hunk 2]
```diff
@@ -180,5 +181,54 @@ public void StopMining_ReturnsSuccess()
...
+            var walletManager = new Mock<IWalletManager>();
+            walletManager.Setup(f => f.GetWalletsNames()).Returns(new List<string> { wallet });
+            walletManager.Setup(f => f.GetAccounts(wallet)).Returns(new List<HdAccount> { new HdAccount { Name = account } });
+
+            this.fullNode.Setup(f => f.NodeFeature<MiningFeature>(false)).Returns(new MiningFeature(this.network, new MinerSettings(Configuration.NodeSettings.Default(this.network)), Configuration.NodeSettings.Default(this.network), this.loggerFactory, new Mock<ITimeSyncBehaviorState>().Object, powMining.Object, null));
+
+            MiningController controller = new MiningController(consensusManager.Object, this.fullNode.Object, this.loggerFactory, this.network, new Mock<IPowMining>().Object, walletManager.Object);
+            WalletAccountReference walletAccountReference = controller.GetAccount();
+
+            Assert.False(string.IsNullOrEmpty(walletAccountReference.AccountName));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs]: Might as well check that it's the right account that is returned.

```suggestion
            Assert.Equal(account, walletAccountReference.AccountName));
```

File: src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs [hunk 3]
```diff
@@ -180,5 +181,54 @@ public void StopMining_ReturnsSuccess()
...
+            Assert.False(string.IsNullOrEmpty(walletAccountReference.WalletName));
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Miner.Tests/Controllers/MiningControllerTest.cs]:
```suggestion
            Assert.Equal(wallet, walletAccountReference.WalletName));
```

**Pull request merged**